WIP Autorag (#20865)
cloudflare/cloudflare-docs: src/content/docs/autorag/get-started.mdx

---

> title: Get started

Reviewer: Screenshots seem like they'd go a long way on this page?

pcx_content_type: get-started
sidebar:
  order: 2
head:
  - tag: title
    content: Get started with AutoRAG
    Description: XX
---

AutoRAG allows developers to create fully managed retrieval-augmented generation (RAG) pipelines to power AI applications with accurate and up-to-date information without needing to manage infrastructure.

## 1. Upload data or use existing data in R2

AutoRAG integrates with R2 for data import. Create an R2 bucket if you don’t have one and upload your data.

:::note
Before you create your first bucket, you must purchase R2 from the Cloudflare dashboard.
:::

To create and upload objects to your bucket from the Cloudflare Dashboard:

1. Log in to the [Cloudflare Dashboard](https://dash.cloudflare.com/?to=/:account/r2) and select **R2**.
2. Select Create bucket, name the bucket, and select **Create bucket**.

> 3. Choose to either drag and drop your file into the upload area or **select from computer**.

Reviewer: Should we give people something to upload?

---
title: Get started
pcx_content_type: get-started
sidebar:
  order: 2
head:
  - tag: title
    content: Get started with AutoRAG
    Description: XX
---

AutoRAG allows developers to create fully managed retrieval-augmented generation (RAG) pipelines to power AI applications with accurate and up-to-date information without needing to manage infrastructure.

> ## 1. (Optional) Upload data

Reviewer: ```suggestion
## 1. Upload data, or use existing data in R2
```

AutoRAG integrates with R2 for data import. Create an R2 bucket if you don’t have one and upload your data. 

:::note
Before you create your first bucket, you must purchase R2 from the Cloudflare dashboard.
:::

To create and upload objects to your bucket from the Cloudflare Dashboard:

1. Log in to the [Cloudflare Dashboard](https://dash.cloudflare.com) and select **R2**.
2. Select Create bucket, name the bucket, and select **Create bucket**.
3. Choose to either drag and drop your file into the upload area or **select from computer**.

## 2. Create an AutoRAG

To create a new AutoRAG:

> 1. Log in to the [Cloudflare Dashboard](https://dash.cloudflare.com) and select **AI** then **AutoRAG**.

Reviewer: You can use `https://dash.cloudflare.com/?to=/:account/stream` format to deep link

2. Select **Create AutoRAG** and complete the setup process:
    - Select a **R2 bucket**.
    - Select an **embedding model**.
    - Select a **LLM model**.
    - Select or create an **AI Gateway**.
    - Enter a **name** for the AutoRAG.
    - Create a new **API token** or reuse an existing API token.
3. Select **Create**

## 3. Monitor indexing 

Once created, AutoRAG will create a Vectorize database in your account and begin indexing the data. 

To monitor the indexing progress:

1. From the **AutoRAG** page in the dashboard, locate and select your AutoRAG.
2. Navigate to the **Overview** page to view the current indexing progress.

## 4. Try it out

> Once indexing is complete, to try out the AutoRAG:

Reviewer: ```suggestion
Once indexing is complete, you can run your first query:
```

1. From the **AutoRAG** page in the dashboard, locate and select your AutoRAG.
2. Navigate to the **Playground** page.
3. Select **Search with AI** or **Search**.
4. Enter a **query** to test out its response.

> ## Use AutoRAG options

Reviewer: Can we be explicit about these as next steps — something like, running RAG queries from your Worker or other application?
